app/src/cose.h

```diff
@@ -60,11 +81,156 @@ namespace scitt::cose
 
   static constexpr const char* SVN_HEADER_PARAM = "svn";
 
+  static std::shared_ptr<ccf::crypto::HashProvider>& get_hash_provider()
+  {
+    static thread_local std::shared_ptr<ccf::crypto::HashProvider>
+      hash_provider;
+    if (!hash_provider)
+    {
+      hash_provider = ccf::crypto::make_hash_provider();
+      if (!hash_provider)
+      {
+        throw std::runtime_error("Failed to create hash provider");
+      }
+    }
+    return hash_provider;
+  }
+
   struct COSEDecodeError : public std::runtime_error
   {
     COSEDecodeError(const std::string& msg) : std::runtime_error(msg) {}
   };
 
+  // Cose Key https://www.rfc-editor.org/rfc/rfc9679.html
+  // Presence of values depends on kty
+  class CoseKeyMap
```

Comment:
Arguably ECCoseKeyMap in its current form.

Reply:
added support for RSA, some explanation in comments